gammapy/modeling/models/spectral.py

```diff
@@ -167,61 +168,83 @@ def __sub__(self, model):
     def __rsub__(self, model):
         return self.__sub__(model)
 
-    def _propagate_error(self, epsilon, fct, **kwargs):
-        """Evaluate error for a given function with uncertainty propagation.
+    def _samples(self, fct, n_samples=10000):
+        """Create SED samples from parameters and covariance
+        using multivariate normal distribution.
 
         Parameters
         ----------
         fct : `~astropy.units.Quantity`
-            Function to estimate the error.
-        epsilon : float
-            Step size of the gradient evaluation. Given as a
-            fraction of the parameter error.
-        **kwargs : dict
-            Keyword arguments.
+            Function to estimate the SED.
+        n_samples : int, optional
+            Number of samples to generate. Default is 10000.
 
         Returns
         -------
-        f_cov : `~astropy.units.Quantity`
-            Error of the given function.
+        sed_samples : np.array
+            Array of SED samples
+
         """
-        eps = np.sqrt(np.diag(self.covariance)) * epsilon
+        rng = np.random.RandomState(seed=42)
```

Comment:
Fixing the seed is dangerous. I understand that this is needed for the test, but otherwise it looks strange.

Reply:
> Fixing the seed is dangerous. I understand that this is needed for the test, but otherwise it looks strange.

I think it's necessary here otherwise the butterfly plots will be slightly different at every run.